R/object_usage_linter.R

```diff
@@ -268,30 +275,26 @@ parse_check_usage <- function(expression,
       function_name,
       capture(
         name = "message",
-        anything,
-        one_of(quote, "\u2018"),
-        capture(name = "name", anything),
-        one_of(quote, "\u2019"),
-        anything
+        zero_or_more(any, type = "lazy"),
+        maybe(
+          one_of(quote, "\u2018"),
+          capture(name = "name", anything),
+          one_of(quote, "\u2019"),
+          anything
+        )
       ),
       line_info
     )
   )
 
+  # nocov start
   missing <- is.na(res$message)
   if (any(missing)) {
-    res[missing, ] <- re_matches(
-      vals[missing],
-      rex(
-        function_name,
-        capture(
-          name = "message",
-          "possible error in ", capture(name = "name", anything), ": ", anything
-        ),
-        line_info
-      )
-    )
+    warning("Possible bug in lintr: Couldn't parse usage message ", sQuote(vals[missing][[1L]]), ". ",
+            "Ignoring ", sum(missing), " usage warnings.")
+    res <- res[!missing, ]
```

Comment:
Left this in as a safety net. We could also go the offensive programming route of deleting the `is.na(res$message)` related part alltogether. WDYT?

Reply:
I like this approach. Maybe leave a TODO to revisit this in "a while" if we don't hear any user feedback?